pkg/virt-launcher/virtwrap/converter/network.go

```diff
@@ -149,6 +162,53 @@ func CreateDomainInterfaces(vmi *v1.VirtualMachineInstance, domain *api.Domain,
 	return domainInterfaces, nil
 }
 
+func newVDPADevicePool(vmi *v1.VirtualMachineInstance) []string {
+	var resources []string
+	var devicePool []string
+
+	ifaces := netvmispec.FilterInterfacesSpec(vmi.Spec.Domain.Devices.Interfaces, func(iface v1.Interface) bool {
+		return iface.VDPA != nil
+	})
+	for _, iface := range ifaces {
+		resourceEnvVarName := fmt.Sprintf("KUBEVIRT_RESOURCE_NAME_%s", iface.Name)
+		resource, isSet := os.LookupEnv(resourceEnvVarName)
+		if !isSet {
+			log.Log.Warningf("%s not set for VDPA interface %s", resourceEnvVarName, iface.Name)
+			continue
+		}
+		resources = append(resources, resource)
+		for _, resource := range resources {
```

Comment:
`resource` is already defined and doesn't need to be defined again here.
```suggestion
		for _, resource = range resources {
```

Reply:
@prnaraya The resource loop should be outside the iface loop. Already fixed.